contrib/platform/src/com/sun/jna/platform/unix/LibCAPI.java

```diff
@@ -223,18 +218,155 @@ public ssize_t(long value) {
+     * must be a multiple of the page size (but length need not be).
+     * @param length The length from the base address to delete mappings. All pages
+     * containing a part of the indicated range are unmapped, and
+     * subsequent references to these pages will generate
+     * {@code SIGSEGV}.
      * @return On success, returns 0. On failure, it returns -1, and {@code errno}
-     *         is set to indicate the cause of the error (probably to
-     *         {@code EINVAL}).
+     * is set to indicate the cause of the error (probably to
+     * {@code EINVAL}).
      */
     int munmap(Pointer addr, size_t length);
+
+    /**
+     * Runs the provided command via "sh" and waits for its termination
+     * (ignoring SIGINT and SIGQUIT, and blocking SIGCHLD).
+     * <p>
+     * This function is a cancellation point in multi-threaded programs. This is a problem if the thread allocates some resources (like memory, file descriptors, semaphores or whatever) at the time system is called.
+     * If the thread gets canceled these resources stay allocated until the program ends. To avoid this calls to system should be protected using cancellation handlers.
+     * <p>
+     * Portability Note: Some C implementations may not have any notion of a command processor that can execute other programs.
+     * You can determine whether a command processor exists by executing system (NULL); if the return value is nonzero, a command processor is available.
+     * <p>
+     * The popen and pclose functions (see Pipe to a Subprocess) are closely related to the system function.
+     * They allow the parent process to communicate with the standard input and output channels of the command being executed.
+     * @param command if null a return value of zero indicates that no command processor is available, otherwise
+     * executes command as a shell command. In the GNU C Library, it always uses the default shell sh to run the command.
+     * In particular, it searches the directories in PATH to find programs to execute.
+     * @return -1 if it wasn't possible to create the shell process, and otherwise is the status of the shell process.
+     * @see <a href="https://www.freebsd.org/cgi/man.cgi?query=system">system(1)</a>
+     */
+    int system(String command);
+
+    /**
+     * @return the process ID of the current process.
+     */
+    int getpid();
+
+    /**
+     * @return the process ID of the parent of the current process.
+     */
+    int getppid();
+
+    /**
+     * Creates a new (child) process.
+     * The current address space is copied over to the new process,
+     * which means that both see {@link #fork()} return, but with different values.
+     *
+     * @return On success, the PID of the child process is returned in the parent, and 0 is returned in the child.
+     * On failure, -1 is returned in the parent, no child process is created, and errno ({@link ErrNo#EAGAIN}, {@link ErrNo#ENOMEM})
```

Comment:
Where is the errno returned? Since on failure -1 is returned in the parent and no child process is created?
Also the status object is kind of weird. More related to this below.